livekit/audio_stream.py

```diff
@@ -30,20 +39,28 @@ def __init__(self, track: Track) -> None:
         stream_info = resp.new_audio_stream.stream
         self._ffi_handle = FfiHandle(stream_info.handle.id)
         self._info = stream_info
-        self._track = track
-        self._queue = ffi_client.subscribe()
+
+        self._task = asyncio.create_task(self._run())
+
+    async def _run(self):
+        while True:
+            event = await self._ffi_queue.wait_for(self._is_event)
+            audio_event = event.audio_stream_event
+
+            if audio_event.HasField('frame_received'):
+                owned_buffer_info = audio_event.frame_received.frame
+                frame = AudioFrame(owned_buffer_info)
+                self._queue.put(frame)
```

Comment:
I'm not sure if it makes more sense to drop already enqueued frames with the ringbuffer vs just dropping new incoming frames with a standard queue that has a capacity.

Also we should log a warning if we ever drop frames

Reply:
my guess is that sometimes it is intended to only want the most recent frames.

Reply:
i don't have a super strong opinion because this is a case where things are failing. Just if we drop frames when we get them (if the queue is full) we don't need to introduce the ring buffer and can more easily log which frames are dropped.

Reply:
for a real-time system, I think we should drop older frames and keep newer ones.